web/genericcom.js

```diff
@@ -23,10 +24,29 @@ if (typeof PDFJSDev !== 'undefined' && !PDFJSDev.test('GENERIC')) {
 
 var GenericCom = {};
 
+class GenericPreferences extends BasePreferences {
+  _writeToStorage(prefObj) {
+    return new Promise((resolve) => {
+      localStorage.setItem('pdfjs.preferences', JSON.stringify(prefObj));
+      resolve();
+    });
+  }
+
+  _readFromStorage(prefObj) {
+    return new Promise((resolve) => {
```

Comment:
Hmm, I'm not sure how I managed to mess that up, but it's fixed now. Thanks for spotting it!